dashboard/src/stores/ingest.ts

```diff
@@ -151,17 +154,34 @@ export const useIngestStore = defineStore("ingest", {
       ]);
     },
     async fetchSips(page: number) {
-      const resp = await client.ingest.ingestListSips({
-        offset: page > 1 ? (page - 1) * this.page.limit : undefined,
-        limit: this.page?.limit || undefined,
-        status: this.filters.status ?? undefined,
-        name: this.filters.name ?? undefined,
-        earliestCreatedTime: this.filters.earliestCreatedTime,
-        latestCreatedTime: this.filters.latestCreatedTime,
-      });
-      this.sips = resp.items;
-      this.page = resp.page;
-      this.updatePager();
+      try {
+        const resp = await client.ingest.ingestListSips({
+          offset: page > 1 ? (page - 1) * this.page.limit : undefined,
+          limit: this.page?.limit || undefined,
+          name: this.filters.name,
+          status: this.filters.status,
+          earliestCreatedTime: this.filters.earliestCreatedTime,
+          latestCreatedTime: this.filters.latestCreatedTime,
+        });
+
+        this.sips = resp.items;
+        this.page = resp.page;
+        this.updatePager();
+      } catch (e) {
+        this.sips = [];
+        this.page = { limit: defaultPageSize, offset: 0, total: 0 };
+        this.updatePager();
+
+        if (e instanceof ResponseError) {
+          const body = await e.response.text();
+          const err = api.ModelErrorFromJSON(JSON.parse(body));
+          console.error("API error response", e.response.status, err.message);
+          throw new Error(err.message);
+        } else {
+          console.error("Unknown error", e);
+          throw new Error("Unknown error when fetching packages");
+        }
+      }
```

Comment:
Instead of a try/catch, I'd prefer `.then()` and `.catch()` over the response promise.

Reply:
Okay, it took me a few tries to get this working, but it's done. :sweat_smile:

Reply:
Yes, I needed to return the promises, that was my problem. :facepalm: